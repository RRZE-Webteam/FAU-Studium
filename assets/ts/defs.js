!function(){"use strict";var t={644:function(t,e,r){Object.defineProperty(e,"__esModule",{value:!0});var n=r(655);n.__exportStar(r(3203),e),n.__exportStar(r(1749),e),n.__exportStar(r(2863),e),n.__exportStar(r(3245),e),n.__exportStar(r(2513),e),n.__exportStar(r(7309),e),n.__exportStar(r(2806),e)},3203:function(t,e){Object.defineProperty(e,"__esModule",{value:!0})},1749:function(t,e){Object.defineProperty(e,"__esModule",{value:!0})},2863:function(t,e){Object.defineProperty(e,"__esModule",{value:!0}),e.DEGREE_ABBREVIATION_GERMAN=e.DEGREE_ABBREVIATION_ENGLISH=void 0,e.DEGREE_ABBREVIATION_ENGLISH={BACHELOR:"B.A.",MASTERS:"M.A.",TEACHING_DEGREE:"B.S.E."},e.DEGREE_ABBREVIATION_GERMAN={BACHELOR:"BA",MASTERS:"MA",TEACHING_DEGREE:"LA"}},3245:function(t,e){Object.defineProperty(e,"__esModule",{value:!0})},2513:function(t,e){Object.defineProperty(e,"__esModule",{value:!0})},7309:function(t,e){Object.defineProperty(e,"__esModule",{value:!0})},2806:function(t,e){Object.defineProperty(e,"__esModule",{value:!0})},655:function(t,e,r){r.r(e),r.d(e,{__assign:function(){return u},__asyncDelegator:function(){return m},__asyncGenerator:function(){return O},__asyncValues:function(){return E},__await:function(){return w},__awaiter:function(){return l},__classPrivateFieldGet:function(){return x},__classPrivateFieldIn:function(){return R},__classPrivateFieldSet:function(){return A},__createBinding:function(){return p},__decorate:function(){return a},__exportStar:function(){return y},__extends:function(){return o},__generator:function(){return s},__importDefault:function(){return g},__importStar:function(){return P},__makeTemplateObject:function(){return j},__metadata:function(){return f},__param:function(){return c},__read:function(){return d},__rest:function(){return i},__spread:function(){return b},__spreadArray:function(){return h},__spreadArrays:function(){return v},__values:function(){return _}});var n=function(t,e){return n=Object.setPrototypeOf||{__proto__:[]}instanceof Array&&function(t,e){t.__proto__=e}||function(t,e){for(var r in e)Object.prototype.hasOwnProperty.call(e,r)&&(t[r]=e[r])},n(t,e)};function o(t,e){if("function"!=typeof e&&null!==e)throw new TypeError("Class extends value "+String(e)+" is not a constructor or null");function r(){this.constructor=t}n(t,e),t.prototype=null===e?Object.create(e):(r.prototype=e.prototype,new r)}var u=function(){return u=Object.assign||function(t){for(var e,r=1,n=arguments.length;r<n;r++)for(var o in e=arguments[r])Object.prototype.hasOwnProperty.call(e,o)&&(t[o]=e[o]);return t},u.apply(this,arguments)};function i(t,e){var r={};for(var n in t)Object.prototype.hasOwnProperty.call(t,n)&&e.indexOf(n)<0&&(r[n]=t[n]);if(null!=t&&"function"==typeof Object.getOwnPropertySymbols){var o=0;for(n=Object.getOwnPropertySymbols(t);o<n.length;o++)e.indexOf(n[o])<0&&Object.prototype.propertyIsEnumerable.call(t,n[o])&&(r[n[o]]=t[n[o]])}return r}function a(t,e,r,n){var o,u=arguments.length,i=u<3?e:null===n?n=Object.getOwnPropertyDescriptor(e,r):n;if("object"==typeof Reflect&&"function"==typeof Reflect.decorate)i=Reflect.decorate(t,e,r,n);else for(var a=t.length-1;a>=0;a--)(o=t[a])&&(i=(u<3?o(i):u>3?o(e,r,i):o(e,r))||i);return u>3&&i&&Object.defineProperty(e,r,i),i}function c(t,e){return function(r,n){e(r,n,t)}}function f(t,e){if("object"==typeof Reflect&&"function"==typeof Reflect.metadata)return Reflect.metadata(t,e)}function l(t,e,r,n){return new(r||(r=Promise))((function(o,u){function i(t){try{c(n.next(t))}catch(t){u(t)}}function a(t){try{c(n.throw(t))}catch(t){u(t)}}function c(t){var e;t.done?o(t.value):(e=t.value,e instanceof r?e:new r((function(t){t(e)}))).then(i,a)}c((n=n.apply(t,e||[])).next())}))}function s(t,e){var r,n,o,u,i={label:0,sent:function(){if(1&o[0])throw o[1];return o[1]},trys:[],ops:[]};return u={next:a(0),throw:a(1),return:a(2)},"function"==typeof Symbol&&(u[Symbol.iterator]=function(){return this}),u;function a(a){return function(c){return function(a){if(r)throw new TypeError("Generator is already executing.");for(;u&&(u=0,a[0]&&(i=0)),i;)try{if(r=1,n&&(o=2&a[0]?n.return:a[0]?n.throw||((o=n.return)&&o.call(n),0):n.next)&&!(o=o.call(n,a[1])).done)return o;switch(n=0,o&&(a=[2&a[0],o.value]),a[0]){case 0:case 1:o=a;break;case 4:return i.label++,{value:a[1],done:!1};case 5:i.label++,n=a[1],a=[0];continue;case 7:a=i.ops.pop(),i.trys.pop();continue;default:if(!(o=i.trys,(o=o.length>0&&o[o.length-1])||6!==a[0]&&2!==a[0])){i=0;continue}if(3===a[0]&&(!o||a[1]>o[0]&&a[1]<o[3])){i.label=a[1];break}if(6===a[0]&&i.label<o[1]){i.label=o[1],o=a;break}if(o&&i.label<o[2]){i.label=o[2],i.ops.push(a);break}o[2]&&i.ops.pop(),i.trys.pop();continue}a=e.call(t,i)}catch(t){a=[6,t],n=0}finally{r=o=0}if(5&a[0])throw a[1];return{value:a[0]?a[1]:void 0,done:!0}}([a,c])}}}var p=Object.create?function(t,e,r,n){void 0===n&&(n=r);var o=Object.getOwnPropertyDescriptor(e,r);o&&!("get"in o?!e.__esModule:o.writable||o.configurable)||(o={enumerable:!0,get:function(){return e[r]}}),Object.defineProperty(t,n,o)}:function(t,e,r,n){void 0===n&&(n=r),t[n]=e[r]};function y(t,e){for(var r in t)"default"===r||Object.prototype.hasOwnProperty.call(e,r)||p(e,t,r)}function _(t){var e="function"==typeof Symbol&&Symbol.iterator,r=e&&t[e],n=0;if(r)return r.call(t);if(t&&"number"==typeof t.length)return{next:function(){return t&&n>=t.length&&(t=void 0),{value:t&&t[n++],done:!t}}};throw new TypeError(e?"Object is not iterable.":"Symbol.iterator is not defined.")}function d(t,e){var r="function"==typeof Symbol&&t[Symbol.iterator];if(!r)return t;var n,o,u=r.call(t),i=[];try{for(;(void 0===e||e-- >0)&&!(n=u.next()).done;)i.push(n.value)}catch(t){o={error:t}}finally{try{n&&!n.done&&(r=u.return)&&r.call(u)}finally{if(o)throw o.error}}return i}function b(){for(var t=[],e=0;e<arguments.length;e++)t=t.concat(d(arguments[e]));return t}function v(){for(var t=0,e=0,r=arguments.length;e<r;e++)t+=arguments[e].length;var n=Array(t),o=0;for(e=0;e<r;e++)for(var u=arguments[e],i=0,a=u.length;i<a;i++,o++)n[o]=u[i];return n}function h(t,e,r){if(r||2===arguments.length)for(var n,o=0,u=e.length;o<u;o++)!n&&o in e||(n||(n=Array.prototype.slice.call(e,0,o)),n[o]=e[o]);return t.concat(n||Array.prototype.slice.call(e))}function w(t){return this instanceof w?(this.v=t,this):new w(t)}function O(t,e,r){if(!Symbol.asyncIterator)throw new TypeError("Symbol.asyncIterator is not defined.");var n,o=r.apply(t,e||[]),u=[];return n={},i("next"),i("throw"),i("return"),n[Symbol.asyncIterator]=function(){return this},n;function i(t){o[t]&&(n[t]=function(e){return new Promise((function(r,n){u.push([t,e,r,n])>1||a(t,e)}))})}function a(t,e){try{(r=o[t](e)).value instanceof w?Promise.resolve(r.value.v).then(c,f):l(u[0][2],r)}catch(t){l(u[0][3],t)}var r}function c(t){a("next",t)}function f(t){a("throw",t)}function l(t,e){t(e),u.shift(),u.length&&a(u[0][0],u[0][1])}}function m(t){var e,r;return e={},n("next"),n("throw",(function(t){throw t})),n("return"),e[Symbol.iterator]=function(){return this},e;function n(n,o){e[n]=t[n]?function(e){return(r=!r)?{value:w(t[n](e)),done:"return"===n}:o?o(e):e}:o}}function E(t){if(!Symbol.asyncIterator)throw new TypeError("Symbol.asyncIterator is not defined.");var e,r=t[Symbol.asyncIterator];return r?r.call(t):(t=_(t),e={},n("next"),n("throw"),n("return"),e[Symbol.asyncIterator]=function(){return this},e);function n(r){e[r]=t[r]&&function(e){return new Promise((function(n,o){(function(t,e,r,n){Promise.resolve(n).then((function(e){t({value:e,done:r})}),e)})(n,o,(e=t[r](e)).done,e.value)}))}}}function j(t,e){return Object.defineProperty?Object.defineProperty(t,"raw",{value:e}):t.raw=e,t}var S=Object.create?function(t,e){Object.defineProperty(t,"default",{enumerable:!0,value:e})}:function(t,e){t.default=e};function P(t){if(t&&t.__esModule)return t;var e={};if(null!=t)for(var r in t)"default"!==r&&Object.prototype.hasOwnProperty.call(t,r)&&p(e,t,r);return S(e,t),e}function g(t){return t&&t.__esModule?t:{default:t}}function x(t,e,r,n){if("a"===r&&!n)throw new TypeError("Private accessor was defined without a getter");if("function"==typeof e?t!==e||!n:!e.has(t))throw new TypeError("Cannot read private member from an object whose class did not declare it");return"m"===r?n:"a"===r?n.call(t):n?n.value:e.get(t)}function A(t,e,r,n,o){if("m"===n)throw new TypeError("Private method is not writable");if("a"===n&&!o)throw new TypeError("Private accessor was defined without a setter");if("function"==typeof e?t!==e||!o:!e.has(t))throw new TypeError("Cannot write private member to an object whose class did not declare it");return"a"===n?o.call(t,r):o?o.value=r:e.set(t,r),r}function R(t,e){if(null===e||"object"!=typeof e&&"function"!=typeof e)throw new TypeError("Cannot use 'in' operator on non-object");return"function"==typeof t?e===t:t.has(e)}}},e={};function r(n){var o=e[n];if(void 0!==o)return o.exports;var u=e[n]={exports:{}};return t[n](u,u.exports,r),u.exports}r.d=function(t,e){for(var n in e)r.o(e,n)&&!r.o(t,n)&&Object.defineProperty(t,n,{enumerable:!0,get:e[n]})},r.o=function(t,e){return Object.prototype.hasOwnProperty.call(t,e)},r.r=function(t){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(t,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(t,"__esModule",{value:!0})};r(644)}();