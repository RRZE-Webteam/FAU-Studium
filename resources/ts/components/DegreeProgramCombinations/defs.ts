import type { Immutable } from 'immer';

import { FormTokenFieldProps } from '@wordpress/components/build-types/form-token-field/types';

export type RelatedDegreeProgramsProps = Readonly< {
	label: string;
	messages: FormTokenFieldProps[ 'messages' ];
	relatedDegreeProgramIds: Immutable< Array< number > >;
	setRelatedDegreePrograms( value: Array< number > ): void;
	/**
	 * Could be used to restrict suggestions by tax query.
	 */
	additionalQuery?: Record< string, unknown >;
} >;
