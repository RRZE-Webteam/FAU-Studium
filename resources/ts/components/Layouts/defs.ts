export interface FormFieldWrapperProps
	extends React.HTMLAttributes< HTMLDivElement > {
	fill?: 'full' | 'half' | 'third' | 'auto';
}
