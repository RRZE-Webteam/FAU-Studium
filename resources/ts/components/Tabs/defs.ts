import { ReactNode } from 'react';

export interface TabItem {
	name: string;

	title: string | JSX.Element;

	className?: string | undefined;

	component: ReactNode;
}
