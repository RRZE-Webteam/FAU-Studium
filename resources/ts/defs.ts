export * from './defs/common';
export * from './defs/core-data';
export * from './defs/degree-program-data';
export * from './defs/entity-selector';
export * from './defs/errors';
export * from './defs/generic';
export * from './defs/server-data';
export * from './defs/term';
