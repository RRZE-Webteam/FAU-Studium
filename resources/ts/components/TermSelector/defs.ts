import { TaxonomySlug, WpTerm } from 'defs/term';

import { FormTokenFieldProps } from '@wordpress/components/build-types/form-token-field/types';

export type Term = WpTerm<any>;

export interface SelectorProps {
    id: string;
    label: string;
    help?: string;
    messages?: FormTokenFieldProps['messages'];
    taxonomy: TaxonomySlug;
}

export interface TermSelectorProps extends SelectorProps {
    value: string;
    onChange: (term: Term | null) => void;
}

export interface MultiTermSelectorProps extends SelectorProps {
    value: string[];
    onChange: (terms: Term[]) => void;
}
