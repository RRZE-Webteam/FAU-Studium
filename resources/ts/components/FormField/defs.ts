import { ReactNode } from 'react';

import { FormFieldWrapperProps } from 'components/Layouts/defs';

import { DegreeProgramDataPaths } from 'defs';

export interface FormFieldProps {
    name: DegreeProgramDataPaths;
    fill?: FormFieldWrapperProps['fill'];
    children: ReactNode;
    label?: string;
    help?: string;
    displayError?: boolean;
    wrapBaseControl?: boolean;
}
