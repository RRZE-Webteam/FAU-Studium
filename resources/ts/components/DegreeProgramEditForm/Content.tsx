import React from 'react';

import { Panel, PanelBody } from '@wordpress/components';

import FormField from 'components/FormField/FormField';
import FormWrapper from 'components/Layouts/FormWrapper';
import { useEditDegreeProgram } from 'contexts/DegreeProgramEditFormProvider';

import ContentField from '../ContentField';
import DegreeProgramCombinations from '../DegreeProgramCombinations';
import MultilingualContainer from './MultilingualContainer';

import { DegreeProgramData, LanguageCode } from 'defs';

const CONTENT_ITEMS = {
    about: 'kurze Beschreibung des Studiengangs allgemein; ±200 Wörter',
    structure: 'Vorstellung des Fachs und der Studieninhalte; ±200 Wörter',
    specializations:
        'Spezialisierungsmöglichkeiten und Schwerpunkte während des Studiums; ±200 Wörter',
    qualities_and_skills:
        'Fähigkeiten und Vorlieben der Studieninteressierten, nicht formelle Voraussetzungen; ±200 Wörter',
    why_should_study:
        'Einzigartigkeit bzw. Besonderheiten (z. B. Kooperationspartner, Verbundstudiumsangebot) des Studiengangs an der FAU; ±200 Wörter',
    career_prospects: 'mögliche Tätigkeitsfelder nach Abschluss des Studiums; ±200 Wörter',
    special_features:
        'Möglichkeiten von Auslandsaufenthalten, Kooperationen mit Forschungsprojekten, etc.; ±200 Wörter',
} as Record<keyof DegreeProgramData['content'], string>;

const REQUIRED_FIELDS = ['about', 'structure'];

const Content = () => {
    const { values, handleChange } = useEditDegreeProgram();

    return (
        <Panel>
            <PanelBody>
                <FormWrapper>
                    {(Object.keys(CONTENT_ITEMS) as Array<keyof typeof CONTENT_ITEMS>).map(
                        (item) => (
                            <FormField
                                key={item}
                                name={`content.${item}.description`}
                                label={`${values.content[item].title.de} ${
                                    values.content[item].title.en
                                        ? `(${values.content[item].title.en})`
                                        : ''
                                }`}
                                help={CONTENT_ITEMS[item]}
                                required={REQUIRED_FIELDS.includes(item)}
                            >
                                <MultilingualContainer value={values.content[item].description}>
                                    {(languageCode: LanguageCode, required) => (
                                        <>
                                            <ContentField
                                                content={
                                                    values.content[item].description[languageCode]
                                                }
                                                onChange={(content) =>
                                                    handleChange<string>(
                                                        `content.${item}.description.${languageCode}`,
                                                        content,
                                                    )
                                                }
                                                required={required}
                                            />
                                        </>
                                    )}
                                </MultilingualContainer>
                            </FormField>
                        ),
                    )}
                    <DegreeProgramCombinations />
                </FormWrapper>
            </PanelBody>
        </Panel>
    );
};

export default Content;
