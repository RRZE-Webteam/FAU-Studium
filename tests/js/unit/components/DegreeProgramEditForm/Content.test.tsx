import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';

import Content from 'components/DegreeProgramEditForm/Content';

import { ContentItem } from 'defs';

const mockContentItem = (label: string): ContentItem => ({
    title: {
        id: '',
        de: label,
        en: label,
    },
    description: {
        id: '',
        de: label,
        en: label,
    },
});

jest.mock('components/DegreeProgramCombinations', () => () => <></>);
jest.mock('contexts/DegreeProgramEditFormProvider', () => ({
    useEditDegreeProgram: () => ({
        values: {
            content: {
                about: mockContentItem('about'),
                structure: mockContentItem('structure'),
                specializations: mockContentItem('specializations'),
                qualities_and_skills: mockContentItem('qualities_and_skills'),
                why_should_study: mockContentItem('why_should_study'),
                career_prospects: mockContentItem('career_prospects'),
                special_features: mockContentItem('special_features'),
                testimonials: mockContentItem('testimonials'),
            },
        },
        handleChange: () => {},
    }),
}));
jest.mock(
    '@wordpress/components',
    jest.fn(() => ({
        BaseControl: jest.fn(({ label, help }) => (
            <>
                <span>{help}</span>
                <span>{label}</span>
            </>
        )),
        Panel: jest.fn(({ children }) => <>{children}</>),
        PanelBody: jest.fn(({ children }) => <>{children}</>),
    })),
);

describe('Content fields', () => {
    afterEach(() => {
        cleanup();
    });

    it('should match snapshot', () => {
        const test = renderer.create(<Content />).toJSON();

        expect(test).toMatchSnapshot();
    });
});
