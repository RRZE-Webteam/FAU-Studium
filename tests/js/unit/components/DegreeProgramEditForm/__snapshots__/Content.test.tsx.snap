// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content fields should match snapshot 1`] = `
[
  <span />,
  <span>
    <span
      className="sc-bcXHqe bKtfDB"
    >
      about (about)
    </span>
    <p
      className="sc-gswNZR hgMeoo"
    >
      kurze Beschreibung des Studiengangs allgemein; ±200 Wörter
    </p>
  </span>,
  <span />,
  <span>
    <span
      className="sc-bcXHqe bKtfDB"
    >
      structure (structure)
    </span>
    <p
      className="sc-gswNZR hgMeoo"
    >
      Vorstellung des Fachs und der Studieninhalte; ±200 Wörter
    </p>
  </span>,
  <span />,
  <span>
    <span
      className="sc-bcXHqe bKtfDB"
    >
      specializations (specializations) (optional)
    </span>
    <p
      className="sc-gswNZR hgMeoo"
    >
      Spezialisierungsmöglichkeiten und Schwerpunkte während des Studiums; ±200 Wörter
    </p>
  </span>,
  <span />,
  <span>
    <span
      className="sc-bcXHqe bKtfDB"
    >
      qualities_and_skills (qualities_and_skills) (optional)
    </span>
    <p
      className="sc-gswNZR hgMeoo"
    >
      Fähigkeiten und Vorlieben der Studieninteressierten, nicht formelle Voraussetzungen; ±200 Wörter
    </p>
  </span>,
  <span />,
  <span>
    <span
      className="sc-bcXHqe bKtfDB"
    >
      why_should_study (why_should_study) (optional)
    </span>
    <p
      className="sc-gswNZR hgMeoo"
    >
      Einzigartigkeit bzw. Besonderheiten (z. B. Kooperationspartner, Verbundstudiumsangebot) des Studiengangs an der FAU; ±200 Wörter
    </p>
  </span>,
  <span />,
  <span>
    <span
      className="sc-bcXHqe bKtfDB"
    >
      career_prospects (career_prospects) (optional)
    </span>
    <p
      className="sc-gswNZR hgMeoo"
    >
      mögliche Tätigkeitsfelder nach Abschluss des Studiums; ±200 Wörter
    </p>
  </span>,
  <span />,
  <span>
    <span
      className="sc-bcXHqe bKtfDB"
    >
      special_features (special_features) (optional)
    </span>
    <p
      className="sc-gswNZR hgMeoo"
    >
      Möglichkeiten von Auslandsaufenthalten, Kooperationen mit Forschungsprojekten, etc.; ±200 Wörter
    </p>
  </span>,
]
`;
